iconservice/icon_service.py

```diff
@@ -85,26 +85,45 @@ async def _serve():
             self._inner_service.clean_close()
 
         loop.create_task(_serve())
-        loop.add_signal_handler(signal.SIGINT, self.close)
-        loop.add_signal_handler(signal.SIGTERM, self.close)
+        loop.add_signal_handler(signal.SIGINT, self.signal_handler, signal.SIGINT)
+        loop.add_signal_handler(signal.SIGTERM, self.signal_handler, signal.SIGTERM)
 
         try:
             loop.run_forever()
         except FatalException as e:
             Logger.exception(e, _TAG)
             Logger.error(e, _TAG)
-            self._inner_service.clean_close()
         finally:
             """
             If the function is called when the operation is not an endless loop 
             in an asynchronous function, the await is terminated immediately.
             """
-            Logger.debug("loop has been stopped and will be closed")
+            Logger.info(f"loop has been stopped and will be closed.")
+
             loop.run_until_complete(loop.shutdown_asyncgens())
+
+            self.cancel_tasks(loop)
+
+            # close icon service components
+            self._inner_service.clean_close()
+
             loop.close()
 
-    def close(self):
-        self._inner_service.clean_close()
+    @staticmethod
+    def cancel_tasks(loop):
+        Logger.info(f"cancel pending tasks in event loop.")
+        pending = asyncio.all_tasks(loop)
```

Comment:
`asyncio.all_tasks()` is new in python3.7.
It would be better to maintain support python3.6 I think.
But we need to discuss about stop to support python3.6.

Reply:
good point.

I will change to `asyncio.Task.all_tasks(loop)`